src/apps/concepts/components/ConceptsActionMenu.tsx

```diff
@@ -0,0 +1,225 @@
+    | "table"
+    | "root"
+    | "paper"
+    | "tableWrapper"
+    | "visuallyHidden"
+    | "buttonLink"
+    | "retired",
+    string
+  >,
+  row: APIConcept,
+  linkedDictionary: string | undefined
+) => {
+  return (
+    <MenuItem onClick={() => toggleMenu(index)}>
+      <Link
+        className={classes.buttonLink}
+        to={`${row.version_url}edit/?linkedDictionary=${linkedDictionary}`}
+      >
+        <EditIcon /> Edit
+      </Link>
+    </MenuItem>
+  );
+};
+
+const addMenu = (
+  index: number,
+  row: APIConcept,
+  buttons: { [key: string]: boolean },
+  toggleMenu: (
+    index: number,
+    event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined
+  ) => void,
+  addConceptsToDictionary: Function
+) => {
+  return !buttons.addToDictionary
+    ? null
+    : addConceptsToDictionaryMenu(
+        addConceptsToDictionary,
+        row,
+        toggleMenu,
+        index
+      );
+};
+
+const removeMenu = (
+  index: number,
+  row: APIConcept,
+  buttons: { [key: string]: boolean },
+  toggleMenu: (
+    index: number,
+    event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined
+  ) => void,
+  linkedSource: string | undefined,
+  removeConceptsFromDictionary: (conceptVersionUrls: string[]) => void
+) => {
+  return !showRemoveFromDictionaryMenuItem(row, buttons.edit, linkedSource)
+    ? null
+    : removeConceptsFromDictionaryMenu(
+        removeConceptsFromDictionary,
+        row,
+        toggleMenu,
+        index
+      );
+};
+
+const editMenu = (
+  index: number,
+  row: APIConcept,
+  buttons: { [key: string]: boolean },
+  classes: Record<
+    | "table"
+    | "root"
+    | "paper"
+    | "tableWrapper"
+    | "visuallyHidden"
+    | "buttonLink"
+    | "retired",
+    string
+  >,
+  toggleMenu: (
+    index: number,
+    event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined
+  ) => void,
+  canModifyConcept: (concept: APIConcept) => boolean,
+  linkedSource: string | undefined,
+  linkedDictionary: string | undefined
+) => {
+  return !showEditMenuItem(row, buttons.edit, linkedSource, canModifyConcept)
+    ? null
+    : editConceptMenu(toggleMenu, index, classes, row, linkedDictionary);
+};
+
+interface ConceptsActionMenuProps {
+  index: number;
+  row: APIConcept;
+  buttons: { [key: string]: boolean };
+  classes: Record<
+    | "table"
+    | "root"
+    | "paper"
+    | "tableWrapper"
+    | "visuallyHidden"
+    | "buttonLink"
+    | "retired",
+    string
+  >;
+  toggleMenu: (
+    index: number,
+    event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined
+  ) => void;
+  menu: { index: number; anchor: null | HTMLElement };
+  canModifyConcept: (concept: APIConcept) => boolean;
+  removeConceptsFromDictionary: (conceptVersionUrls: string[]) => void;
+  addConceptsToDictionary: Function;
+  linkedSource: string | undefined;
+  linkedDictionary: string | undefined;
+}
+
+export function ConceptsActionMenu(props: ConceptsActionMenuProps) {
```

Comment:
@rakesh-1010 @kirity-mahanty can we have sample test cases added for "ConceptsActionMenu" component

Reply:
Added one snapshot test for this component.

Reply:
thanks @kirity-mahanty